src/typing_manager.ts

```diff
@@ -35,19 +35,62 @@ export class TypingManager implements Disposable {
     private composingText = "";
 
     /**
-     * Flag indicating that we should take over vscode input
-     * If false, we should forward all input received from "type" to "default:type"
+     * Flag indicating that we should take over vscode input, where "take over"
+     * here means handling input by nvim.
+     * If false, we should forward all input received from "type" to "default:type",
+     * and "replacePreviousChar" to "default:replacePreviousChar".
      */
-    private takeOverVSCodeInput = false;
+    private _takeOverVSCodeInput = false;
 
     // configs
+    private useCompositeKeys!: boolean;
     private compositeKeys!: CompositeKeys;
     private compositeFirstKeys!: string[];
     private compositeSecondKeysForFirstKey!: Map<string, string[]>;
     // logic variables
     private compositeMatchedFirstKey?: string;
     private compositeTimer?: NodeJS.Timeout;
 
+    // Notes:
+    // 1. "type" and "replacePreviousChar" must be registered at the same time
+    // 2. Forwarding the arguments of replacePreviousChar to default:replacePreviousChar
+    //    causes text jitter during ime composition
+    //
+    // The compromise solution used here:
+    // 1. When composite keys are needed, always register type and
+    //    replacePreviousChar at the same time, which unavoidably causes text
+    //    flickering during ime composition.
+    // 2. When composite keys are not needed, only register type and
+    //    replacePreviousChar when it's necessary to take over vscode input.
+
+    private typeHandler?: Disposable;
+    private replacePreviousCharHandler?: Disposable;
+
+    private get takeOverVSCodeInput() {
+        return this._takeOverVSCodeInput;
+    }
+
+    private set takeOverVSCodeInput(takeOver: boolean) {
+        this._takeOverVSCodeInput = takeOver;
+
+        if (takeOver) {
+            if (!this.typeHandler) this.typeHandler = commands.registerTextEditorCommand("type", this.onVSCodeType);
+            if (!this.replacePreviousCharHandler)
+                this.replacePreviousCharHandler = commands.registerCommand(
+                    "replacePreviousChar",
```

Comment:
why *register* the command every time `takeOverVSCodeInput=true` is assigned? I would think that the command(s) can be registered once, and there implementation can check state to decide what to do.

Reply:
>     // Notes:
>     // 1. "type" and "replacePreviousChar" must be registered at the same time
>     // 2. Forwarding the arguments of replacePreviousChar to default:replacePreviousChar
>     //    causes text jitter during ime composition
>     //
>     // The compromise solution used here:
>     // 1. When composite keys are needed, always register type and
>     //    replacePreviousChar at the same time, which unavoidably causes text
>     //    flickering during ime composition.
>     // 2. When composite keys are not needed, only register type and
>     //    replacePreviousChar when it's necessary to take over vscode input.

Reply:
oh, I see that this is overwriting a vscode builtin: https://github.com/microsoft/vscode/blob/3982027f02dee5aa52622f7f2812c70f04ce337a/src/vs/editor/browser/coreCommands.ts#L2167

it sure would be worth mentioning that in a code comment somewhere.

Reply:
I did read that comment, but it doesn't mention that this is a vscode builtin, which is critical for understanding this.

Reply:
We can mention [vim-sample](https://github.com/microsoft/vscode-extension-samples/tree/main/vim-sample) in CONTRIBUTING.md

Reply:
sure, but that still leaves other developers with a puzzle to piece together. It costs nearly zero to mention in this context that "replacePreviousChar" is a vscode builtin.

Reply:
Oh, the relevant content has been explained here: https://github.com/vscode-neovim/vscode-neovim/blob/master/CONTRIBUTING.md#typingmanager

Reply:
Again, we are talking about a 1-line addition that is *near* this code.

Reply:
What's the relation between this and this PR? More than just these two vscode commands are overridden.

Reply:
This is something I noticed while reviewing this PR. It's a zero-cost improvement that can be made, regardless of other possible, future improvements. Is any other justification needed?